src/components/CreateCampaign/StepOne/BannerSizesList.tsx

```diff
@@ -0,0 +1,51 @@
+import { Grid } from '@mantine/core'
+import BannerSizeMock from 'components/common/BannerSizeMock'
+import { checkBannerSizes, checkSelectedDevices } from 'helpers/createCampaignHelpers'
+import useCreateCampaignContext from 'hooks/useCreateCampaignContext'
+import { useMemo } from 'react'
+import { AdUnit } from 'adex-common/dist/types'
+
+const BannerSizesList = ({ adUnits }: { adUnits: AdUnit[] }) => {
+  const {
+    campaign: { devices }
+  } = useCreateCampaignContext()
+  const selectedDevices = useMemo(() => checkSelectedDevices(devices), [devices])
+  const updatedBannerSizes = useMemo(() => checkBannerSizes(adUnits), [adUnits])
+
+  const mobileSizes = useMemo(
+    () => updatedBannerSizes.filter((item) => item.device === 'mobile'),
+    [updatedBannerSizes]
+  )
+  const desktopSizes = useMemo(
+    () => updatedBannerSizes.filter((item) => item.device === 'desktop'),
+    [updatedBannerSizes]
+  )
+
+  return selectedDevices === 'mobile' ? (
+    <Grid>
+      {mobileSizes.map((item) => (
+        <Grid.Col span="content" key={`${item.bannerSizes.w}x${item.bannerSizes.h}`}>
+          <BannerSizeMock variant={item} />
+        </Grid.Col>
+      ))}
+    </Grid>
+  ) : selectedDevices === 'desktop' ? (
+    <Grid>
+      {desktopSizes.map((item) => (
+        <Grid.Col span="content" key={`${item.bannerSizes.w}x${item.bannerSizes.h}`}>
+          <BannerSizeMock variant={item} />
+        </Grid.Col>
+      ))}
+    </Grid>
+  ) : selectedDevices === 'both' ? (
+    <Grid>
+      {updatedBannerSizes.map((item) => (
+        <Grid.Col span="content" key={`${item.bannerSizes.w}x${item.bannerSizes.h}`}>
+          <BannerSizeMock variant={item} />
+        </Grid.Col>
+      ))}
+    </Grid>
+  ) : null
+}
```

Comment:
This uses too many ternary operators and all of them are conditional the last else that is null will never execute, this can be replaced with:

if (selectedDevices === 'desktop') return desktop;
if (selectedDevices === 'mobile') return mobile;
return both;

if used like this the component is more readable and maintainable.